<?xml version="1.0"?>
<launch>
  <arg name="device"    default="/dev/video0"/>
  <arg name="width"     default="1280"/>
  <arg name="height"    default="720"/>
  <!--arg name="framerate" default="15"/-->
  <arg name="input_topic" default="/theta_s_uvc/image_raw"/>
  <arg name="output_topic" default="image_out"/>
  <group ns="dual_fisheye_blending">
    <node name="left_dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" args="load reprojection $(find reprojection)/params/dual_fisheye.yaml" />
    <!-- <node name="right_dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" args="load right_reprojection $(find reprojection)/params/dual_fisheye_right.yaml" />
    <node name="rotated_dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" args="load rotated_reprojection $(find reprojection)/params/dual_fisheye_left.yaml" /> -->
    <node pkg="reprojection" type="dual_fisheye_blending"
      name="dual_fisheye_blending" output="screen">
      <param name="width"           	value="$(arg width)"/>
      <param name="height"          	value="$(arg height)"/>
      <param name="vertex_shader" 	value="$(find opengl_ros_lib)/shader/vs_passthrough.glsl"/>
      <param name="reprojection_fragment_shader" 	value="$(find reprojection)/shader/reprojection.frag"/>
      <param name="blending_fragment_shader" 	value="$(find opengl_ros)/shader/blend_images.glsl"/>
      <param name="image_in" 		value="$(arg input_topic)"/>
      <param name="image_out" 		value="/blended/image"/>
    </node>
  </group>

  <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="reconfigure_gui"/> 

  <node pkg="image_view" type="image_view" name="viewer_in">
    <remap from="image" to="$(arg input_topic)"/>
    <param name="window_name" value="Dual Fisheye"/>
  </node>

  <!--node pkg="image_view" type="image_view" name="left_equi_viewer">
    <remap from="image" to="/left_equi/image"/>
    <param name="window_name" value="Left Equi"/>
  </node>

  <node pkg="image_view" type="image_view" name="right_equi_viewer">
    <remap from="image" to="/right_equi/image"/>
    <param name="window_name" value="Right Equi"/>
  </node>
  
  <node pkg="image_view" type="image_view" name="right_equi_back_viewer">
    <remap from="image" to="/right_equi_rotated/image"/>
    <param name="window_name" value="Right Equi Back"/>
  </node-->

  <node pkg="image_view" type="image_view" name="blended_viewer">
    <remap from="image" to="/blended/image"/>
    <param name="window_name" value="Blended Equi Image"/>
  </node>
</launch>
